### ISSUE TITLE
Nonsensical special types are uncaught by `check-types`

### ISSUE THREAD

**Issue by teddy-error** [gajus/eslint-plugin-jsdoc#416]
Recently we observed a developer, who was unfamiliar with JSDoc, attempt to define a type like this:

```js
  /**
   * @param {Object<Object>}
   */
```

I thought it odd that this was not detected by the linter. I checked if this was a special case but found that there was no such check performed on special types because

```js
  /**
   * @param {string<Object>}
   */
```
is also passed. 👎 

```js
  /**
   * @param {string<>}
   */
```
however *is* invalidated because the subtype is nonexistent. 👍 

AFAIK special types apply in at least two cases:

1. Array<type> and
2. Promise<resolvedType>

would it be possible to catch any type defined outside of these?

**Comment by brettz9** [comment]
Good finds... Another allowable case is with generics; see https://github.com/google/closure-compiler/wiki/Generic-Types#multiple-template-types (with jsdoc explicitly supporting all Closure types and TypeScript adopting generics as well).

But yes, this seems too permissive. I've filed https://github.com/jsdoctypeparser/jsdoctypeparser/issues/99 since I think this should be disallowed at the type parser level. I'm therefore closing this issue here, but feel free to subscribe to that issue.

**Comment by teddy-error** [comment]
Thanks!
